src/main/java/org/apache/commons/lang3/time/FastTimeZone.java

```diff
@@ -84,8 +84,9 @@ private static int parseInt(final String group) {
         return group != null ? Integer.parseInt(group) : 0;
     }
 
+    @SuppressWarnings("argument.type.incompatible")
     private static boolean parseSign(final String group) {
-        return group != null && group.charAt(0) == '-';
+        return group != null && group.charAt(0) == '-'; // if group == null, group.charAt(0) is not evaluated
```

Comment:
`group` could still have length of zero even if it is non-null, which is what the typechecker is complaining about. I would annotated `group` as `@MinLen(1) String` instead of suppressing this warning.